### PR TITLE
Regression fixes: LLM logging; client readiness (EventStreamRuntime)

### PR DESCRIPTION
**Pull request by tobitege**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Fix logging of LLM messages (if enabled) and client readiness detection in EventStreamRuntime

---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

- Unfortunately a recent PR of mine broke the logging of LLM messages to file, which caused integration test mock files to miss the user prompt for some agents+test combos.
  EDIT: ~~Newer PR's since then started to fail.~~ The regeneration of integration mock files for PR's was problematic.
- For the `EventStreamRuntime` the client readiness detection was moved to the actual `LogBuffer` class. In some situations, depending on timing, OH could've gotten stuck on getting docker container logs. This detection was added to prevent premature sending of commands before the client was fully initialized (a container is signaled as ready several seconds before that).
- Several integration test mock files were regenerated for this PR.

---

**Link of any specific issues this addresses**
#3665

### CODE REVIEW

**Comment by enyst** [comment]
> the logging of LLM messages to file, which caused integration test mock files to miss the user prompt for some agents+test combos. Newer PR's since then started to fail.

How did CI pass in order to merge if they fail? I see PRs succeeding CI, can you please point out which fails?

**Comment by tobitege** [comment]
> > the logging of LLM messages to file, which caused integration test mock files to miss the user prompt for some agents+test combos. Newer PR's since then started to fail.
> 
> How did CI pass in order to merge if they fail? I see PRs succeeding CI, can you please point out which fails?

My bad, I wrote that incorrectly: trying to regenerate for PR's would fail. I reproduced that multiple times in my WSL with e.g. CodeActAgent and the test_ipython test where the user prompt was not in the message logged in the prompts.

File: openhands/llm/llm.py [hunk 1]
```diff
@@ -236,11 +212,11 @@ def wrapper(*args, **kwargs):
...
             message_back = resp['choices'][0]['message']['content']
+            if message_back:
+                llm_response_logger.debug(message_back)
 
-            llm_response_logger.debug(message_back)
-
-            # post-process to log costs
-            self._post_completion(resp)
+                # post-process to log costs
+                self._post_completion(resp)
```
Comment (xingyaoww) [inline openhands/llm/llm.py:219]: I think even if the `message_back` is empty string - we might still incur costs (eg input tokens), right? I'd suggest we move this line out of the if statement.

```suggestion
            self._post_completion(resp)
```

**Review by xingyaoww — APPROVED** [review]
LGTM!

File: openhands/llm/llm.py (hunk 1 above)
Comment (tobitege) [reply openhands/llm/llm.py:219]: Here's why I put that in on purpose: I saw integration tests having 2 completions, but 3 steps and thus fail

Comment (xingyaoww) [reply openhands/llm/llm.py:219]: If that's the case, let's leave this as is then... (until it cause some error in the future)

**Pull request merged**